pennylane/qnn/keras.py

```diff
@@ -318,8 +311,14 @@ def _evaluate_qnode(self, x):
             **{k: 1.0 * w for k, w in self.qnode_weights.items()},
         }
         res = self.qnode(**kwargs)
+
         if isinstance(res, (list, tuple)):
+            if len(tf.shape(x)) > 1:
+                # multi-return and batch dim case
+                res = [tf.reshape(r, (tf.shape(x)[0], -1)) for r in res]
```

Comment:
So for mixed return types, this step concatenates them into a flat array, do I understand this correctly?
This approach differs from the (new) return type system of PennyLane, do we care about this fact? I am not too familiar with keras layers, but I suppose it is required that it returns a flat output dimension.

Reply:
To be clear: This is not a change made by this PR, the `hstack` was there before, so I'm just curious whether we are aware of this inconsistency. :)

Reply:
Yes I am not sure you can return anything else than a single dimension tf tensor here?

Reply:
The resulting tensor (after `hstack`ing) has shape `(batch_size, output_dim)`. This is consistent with what we expect with parameter broadcasting.

I believe the previous implementation (with `hstack` just by itself) has a bug. If, for example, the user passes in a batch size of 4 and the QNode has two `expval` measurements, then the previous implementation would return an array of shape `(8,)` instead of `(4, 2)`.

Reply:
In the return types system, this would lead to a `tuple` being returned, but I guess that is not really desired when working with a `Layer` as part of a QNN...

Reply:
Nope, we definitely want to return a single `tf.Tensor`, but it's still up in the air what the behaviour should be when combining measurements with incompatible shapes.

Reply:
I agree with you @eddddddy, that would be a good solution, we cannot have the return system from PennyLane there. This post processing makes sense because we are using external libraries and have to adapt to their need.

Reply:
This sounds very reasonable to me. :+1:

Reply:
Changed and added docs.